WordPress/Classes/ViewRelated/Me/MyProfileViewController.swift

```diff
@@ -7,19 +7,52 @@ func MyProfileViewController(account: WPAccount) -> ImmuTableViewController? {
     }
 
     let service = AccountSettingsService(userID: account.userID.intValue, api: api)
-    return MyProfileViewController(service: service)
+    let headerView = makeHeaderView(account: account)
+    return MyProfileViewController(account: account, service: service, headerView: headerView)
 }
 
-func MyProfileViewController(service: AccountSettingsService) -> ImmuTableViewController {
-    let controller = MyProfileController(service: service)
+func MyProfileViewController(account: WPAccount, service: AccountSettingsService, headerView: MyProfileHeaderView) -> ImmuTableViewController {
+    let controller = MyProfileController(account: account, service: service, headerView: headerView)
     let viewController = ImmuTableViewController(controller: controller)
+    headerView.onAddUpdatePhoto = {
+        controller.presentGravatarPicker(viewController)
```

Comment:
There's a retain cycle in this closure – the closure retains the controller and the view controller, but is in turn retained by the header view, which is retained by the controller. It's a bit of an odd situation due to the _unique_ nature of `MyProfileViewController`. I think the fix is just to make the closure's capture of the controller and view controller weak:

```
headerView.onAddUpdatePhoto = { [weak controller, weak viewController] in
    if let viewController = viewController {
        controller?.presentGravatarPicker(viewController)
    }
}
```